### PR TITLE
Header

### PR DESCRIPTION
**Pull request by ahmadjarafat**
- Create Header
- Make header responsive for all screen sizes

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -3,23 +3,16 @@ import theme from "./themes/theme";
...
     <BrowserRouter>
       <ThemeProvider theme={theme}>
-        {/* <Header /> */}
-        <HeroSection/>
-        <NewArrivalsSection/>
-        <BrandsSection/>
-        <Banner/>
-        <Footer/>
+        <Header />
+        <Home />
```
Comment (Rahaf-96) [inline src/App.jsx:14]: you removed these components from App, but I can't find them added in Home component.

 ```
<HeroSection/>
        <NewArrivalsSection/>
        <BrandsSection/>
        <Banner/>
```

File: src/components/header/index.jsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+import userIcon from "../../assets/icons/user.svg";
+import bagIcon from "../../assets/icons/bag.svg";
+import NavLink from "../links/nav-link";
+import useMediaQuery from "@mui/material/useMediaQuery";
+import ButtonWithIcon from "../buttons/button-with-Icon";
+import MenuIcon from "@mui/icons-material/Menu";
+import Drawer from "@mui/material/Drawer";
+import { useState } from "react";
+
+const Header = ({ Categories }) => {
```
Comment (Rahaf-96) [inline src/components/header/index.jsx]: name props using camelCase please

File: src/components/header/index.jsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+            return !isMobile ? (
+              <IconButton key={index} sx={{ padding: "0px" }}>
+                <img src={icon} />
+              </IconButton>
+            ) : null;
+          })}
+        </Stack>
+      </Stack>
+      <Drawer
+        // container={container}
```
Comment (Rahaf-96) [inline src/components/header/index.jsx]: remove unused code please. or mark it as a TODO if needed

File: src/components/header/index.jsx (hunk 2 above)
Comment (Rahaf-96) [inline src/components/header/index.jsx]: do you want to use this prop (categories), inside your code?
As I see it's not used

File: src/components/header/index.jsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+import MenuIcon from "@mui/icons-material/Menu";
+import Drawer from "@mui/material/Drawer";
+import { useState } from "react";
+
+const Header = ({ Categories }) => {
+  const isLaptop = useMediaQuery((theme) => theme.breakpoints.down("lg"));
+  const isTablet = useMediaQuery((theme) => theme.breakpoints.down("md"));
+  const isMobile = useMediaQuery((theme) => theme.breakpoints.down("sm"));
+
+  const [mobileOpen, setMobileOpen] = useState(false);
```
Comment (Rahaf-96) [inline src/components/header/index.jsx]: it's a good practice to start boolean variables names with is or has.
for example: isMobileOpen, setIsMobileOpen

File: src/components/header/index.jsx [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    setMobileOpen((prevState) => !prevState);
+  };
+
+  const drawer = (
+    <List>
+      <ListItem>
+        <ListItemText primary={"Categories"} />
+      </ListItem>
+      <Divider />
+      {["Handbags", "Watches", "Skincare", "Jewellery", "Apperels"].map(
```
Comment (Rahaf-96) [inline src/components/header/index.jsx]: It's better to store this array as a constant then use it here.

**Review by Rahaf-96 — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 1 above)
Comment (ahmadjarafat) [reply src/App.jsx:14]: Zaid will put the home page together and create the routing, I only put the components that I created

File: src/components/header/index.jsx (hunk 2 above)
Comment (ahmadjarafat) [reply src/components/header/index.jsx]: Done

File: src/components/header/index.jsx (hunk 3 above)
Comment (ahmadjarafat) [reply src/components/header/index.jsx]: Done

File: src/components/header/index.jsx (hunk 2 above)
Comment (ahmadjarafat) [reply src/components/header/index.jsx]: Done

File: src/components/header/index.jsx (hunk 4 above)
Comment (ahmadjarafat) [reply src/components/header/index.jsx]: Done

File: src/components/header/index.jsx (hunk 5 above)
Comment (ahmadjarafat) [reply src/components/header/index.jsx]: Done

**Review by Rahaf-96 — APPROVED** [review]

**Pull request merged**